proxy/http2/Http2ConnectionState.cc

```diff
@@ -936,30 +940,70 @@ Http2ConnectionState::cleanup_streams()
 void
 Http2ConnectionState::delete_stream(Http2Stream *stream)
 {
+  // The following check allows the method to be called safely on already deleted streams.
+  if (deleted_from_active_streams(stream)) {
+    return;
+  }
+
```

Comment:
OK, needed that for calling `delete_stream()` on already deleted streams for my “catch-all-delete-stream” calls (6.2.1). 

Also grouped some of the `DLL<>` + `client_streams_count` updates in separate functions.

Looking at how the master changed since I started working on this I may need to revert that to be consistent with master.